servers/src/grin/seed.rs

```diff
@@ -176,6 +177,13 @@ fn monitor_peers(
 		None => debug!(LOGGER, "monitor_peers: no preferred peers"),
 	}
 
+	// take a random defunct peer and mark it healthy: over a long period any
```

Comment:
How about just retry connecting, instead of indirectly change the peer state.

i.e.  move this paragraph to the end of this function, and 
```
	// try connecting to a random defunct peer: over a long period any peer will see another as defunct eventually,
	// gives us a chance to retry
	if defuncts.len() > 0 {
		thread_rng().shuffle(&mut defuncts);
		tx.send(defuncts[0].addr).unwrap();
	}
```

Reply:
I found that to be perhaps less elegant because you'd also need to check whether the maximum peer counts is  exceeded. And if you're at max peer count would never try any resurrection (because it'd be ditched from the channel), which could still be damaging longer term. Just a simple update seemed to mesh better with existing logic.